File: Tienda.py
class Tienda:

    def __init__(self):
        self.usuarios = []
        self.productos = []
        self.total_ventas = 0

    def agregar_producto_a_carrito(self, usuario, producto, cantidad):
        if producto.hay_unidades(cantidad):
            usuario.agregar_producto_a_carrito(producto, cantidad)
        else:
            print(f"No hay suficientes unidades disponibles de {producto.nombre}, solo quedan {producto.unidades_disponibles} unidades")

    def terminar_compra(self, usuario):
        for item in usuario.carrito.items:
            item.producto.restar_unidades(item.cantidad)
            self.total_ventas += usuario.carrito.valor_total_venta

File: Backend/model/Tienda.ts
import Producto from "./Producto";

export default class Tienda{

    private productos: Producto[];
    private totalVentas: number;

    constructor(productos: Producto[], totalVentas: number){
        this.productos = productos;
        this.totalVentas = totalVentas;
    }

    public getTotalVentas(){
        return this.totalVentas;
    }

    public getProductos(): Producto[]{
        return this.productos;
    }

    public sumarVenta(venta:number){
        this.totalVentas += venta;
    }

    public restarUnidadesDeProducto(sku: string, cantidad: number){
        this.productos = this.productos.map(producto => {
            if(producto.sku === sku){
                producto.restarUnidades(cantidad)
            }
            return producto;
        })
    }

}